### PR TITLE
feat: add status prop in checkbox, radio, select, slider and toggle

### PR DESCRIPTION
**Pull request by gepd**
## Checklist

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

To make components more consistents, added the `status` prop who allows to show the component with different colors based in a status.
components affected: `checkbox`, `radio`, `select`, `slider` and `toggle`

Will solve https://github.com/geist-org/react/issues/507

### CODE REVIEW

File: components/checkbox/checkbox.icon.tsx [hunk 1]
```diff
@@ -4,16 +4,17 @@ import useTheme from '../use-theme'
 interface Props {
   disabled?: boolean
   checked?: boolean
+  fill?: string
+  bg?: string
 }
 
-const CheckboxIcon: React.FC<Props> = ({ disabled, checked }) => {
+const CheckboxIcon: React.FC<Props> = ({ fill, bg, disabled, checked }) => {
```
Comment (unix) [inline components/checkbox/checkbox.icon.tsx:11]: `({ fill: propsFill, bg: propsBg ... })`
It is recommended not to use `_`.

File: components/select/select.tsx [hunk 2]
```diff
@@ -10,12 +10,13 @@ import SelectDropdown from './select-dropdown'
...
 import Grid from '../grid'
 import { SelectContext, SelectConfig } from './select-context'
-import { getSizes } from './styles'
+import { getColors, getSizes } from './styles'
 import Ellipsis from '../shared/ellipsis'
 
 interface Props {
   disabled?: boolean
   size?: NormalSizes
+  status?: NormalTypes
```
Comment (unix) [inline components/select/select.tsx:19]: In the `Dot`, `Badge` and other components, Geist is now using `type` to specify the status, can we change the `status` to `type`? Consistent interface with other components.

File: pages/en-us/components/select.mdx [hunk 3]
```diff
@@ -26,6 +26,39 @@ Display a dropdown list of items.
 `}
 />
 
+<Playground
+  title="status"
+  desc="Differentiate states by color."
+  scope={{ Select, Spacer }}
+  code={`
+() => {
+  return (
```
Comment (unix) [inline pages/en-us/components/select.mdx:35]: `return` is not necessary.

**Review by unix — CHANGES_REQUESTED** [review]
Looks very good! 
There are still some details that will take you some more time, so if you solve them please at me, I need to debug and review them locally and clean up some commits in the meantime

**Comment by gepd** [comment]
@unix  all Done!

**Comment by gepd** [comment]
Hey @unix I was using the modal compontent and I realised that the actions doesn't have the `type` property, I've a commit fixing that, but I've a doubt with the `passive` attribute, I have completely removed it as I think the `type` prop would be enoght, do you agree?

**Review by unix — APPROVED** [review]

**Pull request merged**